### PR TITLE
Update WinForms assembly TFM to net9.0

### PR DESCRIPTION
**Pull request by dreddy-work**
Updating TFM for WinForms assemblies.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10108)

### CODE REVIEW

**Comment by dreddy-work** [comment]
@singhashish-wpf / @pchaurasia14, we need TFM change in WPF as well to get this flowed smoothly.

**Review by lonitra — APPROVED** [review]

**Review by mmitche — APPROVED** [review]

File: Directory.Build.targets [hunk 1]
```diff
@@ -40,7 +40,7 @@
     <_ProjectDirectoryRelativePath>$([MSBuild]::MakeRelative($(RepoRoot), $(MSBuildProjectDirectory)))</_ProjectDirectoryRelativePath>
     <_IsUnderTestsFolder>$(_ProjectDirectoryRelativePath.Contains("\tests\"))</_IsUnderTestsFolder>
     <IsShipping Condition="$(_IsUnderTestsFolder)">false</IsShipping>
-	<RedistTargetFrameworkName>$(TargetFrameworkName)$(TargetFrameworkMajorVersion).</RedistTargetFrameworkName>
+	<RedistTargetFrameworkName>$(NetCurrent)</RedistTargetFrameworkName>
```
Comment (ViktorHofer) [inline Directory.Build.targets:43]: Are you sure this is right? This was `net8.` before and now is `net9.0`.

Comment (ViktorHofer) [reply Directory.Build.targets:43]: Ah yeah, I just checked all conditions. This was done to condition on all minor releases as well which we don't do anymore.

Comment (dreddy-work) [reply Directory.Build.targets:43]: Right.

**Pull request merged**